src/v/datalake/coordinator/types.h

```diff
@@ -27,6 +27,7 @@ enum class errc : int16_t {
     stale,
     concurrent_requests,
     revision_mismatch,
+    other,
```

Comment:
Curious if this gets used? I could be wrong but it seems like coordinator::cooridnator::errc is only logged, never returned to an RPC?

Reply:
It is not used anywhere, but it was needed for the mapping of coordinator-specific error codes (`to_rpc_errc` function). I don't see existing error codes here that new coordinator errc's could be mapped to. Or do you think we should add some catch-all code here?